src/common/vector/value_vector.cpp

```diff
@@ -7,14 +7,27 @@ namespace common {
 
 ValueVector::ValueVector(DataType dataType, storage::MemoryManager* memoryManager)
     : dataType{std::move(dataType)} {
-    valueBuffer = std::make_unique<uint8_t[]>(
-        Types::getDataTypeSize(this->dataType) * DEFAULT_VECTOR_CAPACITY);
+    numBytesPerValue = this->dataType.typeID == VAR_LIST ? sizeof(common::ListEntry) :
```

Comment:
Leave a TODO to move this logic to Types::getDataTypeSize once we remove ku_list

Reply:
done